lib/rules/no-dupe-keys.js

```diff
@@ -63,12 +105,23 @@ module.exports = {
       }),
       utils.defineScriptSetupVisitor(context, {
         onDefinePropsEnter(node, props) {
+          const propsNode = getPropsPattern(node)
+          const propReferences = [
+            ...(propsNode ? extractReferences(propsNode, context) : []),
+            node
+          ]
+
           for (const prop of props) {
             if (!prop.propName) continue
 
             const variable = findVariable(context.getScope(), prop.propName)
             if (!variable || variable.defs.length === 0) continue
 
+            for (const def of variable.defs) {
+              if (def.type !== 'Variable') continue
+              if (isInsideInitializer(def.node, propReferences)) return
```

Comment:
Is it really `return` in this case or should it also be `continue`?

Reply:
This is correct. Ignore the report if the props variable is included in the variable initialization.

Reply:
Sorry. This is incorrect. The outer loop should continue.

Reply:
I fixed that. Thanks for finding that mistake!

Reply:
Thanks for adding a test case!